In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns

In [ ]:
from google.colab import files
data=pd.read_excel('nba2.xlsx')

In [ ]:
data.head()

DATASET  R/P        DATE  ... REST DAYS  OPENING ODDS  SPREAD
0  2016-2017 Regular Season    1  10/25/2016  ...         3         205.5     9.5
1  2016-2017 Regular Season    1  10/25/2016  ...         3          -9.5    -9.5
2  2016-2017 Regular Season    1  10/25/2016  ...         3         194.0     6.0
3  2016-2017 Regular Season    1  10/25/2016  ...         3          -6.0    -6.0
4  2016-2017 Regular Season    1  10/25/2016  ...         3         210.5     8.0

[5 rows x 43 columns]

In [ ]:
data.describe()

R/P          H/R          W/E  ...    REST DAYS  OPENING ODDS       SPREAD
count  2460.0  2460.000000  2460.000000  ...  2460.000000   2460.000000  2460.000000
mean      1.0     1.500000     1.500000  ...     1.037398    102.493902     0.000000
std       0.0     0.500102     0.500102  ...     0.735593    108.452662     6.829624
min       1.0     1.000000     1.000000  ...     0.000000    -22.000000   -22.000000
25%       1.0     1.000000     1.000000  ...     1.000000     -5.000000    -5.000000
50%       1.0     1.500000     1.500000  ...     1.000000     90.000000     0.000000
75%       1.0     2.000000     2.000000  ...     1.000000    210.500000     5.000000
max       1.0     2.000000     2.000000  ...     3.000000    241.000000    22.000000

[8 rows x 40 columns]

第一題

In [ ]:
#平均得分
data['PTS'].mean()

105.59024390243903

In [ ]:
#平均3分球命中數
data['3P'].mean()

9.653658536585366

In [ ]:
#平均籃板球數
data['TOT'].mean()

43.510569105691054

#第一題分析 
NBA2016-2017賽季(例行賽)
1.   平均得分:大約105.6分
2.   平均3分球命中數:9.65 約9~10顆
3.   平均籃板球數:43.51 約43~44顆     



第二題

In [ ]:
data.groupby('H/R').mean()[['W/L','PTS','3P','TOT']]

W/L         PTS        3P        TOT
H/R                                          
1    0.58374  107.165041  9.898374  44.083740
2    0.41626  104.015447  9.408943  42.937398

#第二題分析 
NBA2016-2017賽季(例行賽)

1為主場 2為客場

主場勝率比客場高

主場平均分數比客場高約3%

主客場的三分球命中數差不多 大約相差0.48%

主場的籃板球數比客場高大約1%

以上總結:在主場時的分數會比客場好，勝率也比客場高